docs/guides/datamigrations/postgres.rst

```diff
@@ -0,0 +1,581 @@
+        }
+        required description: str;
+    }
+
+    type Status {
+        required title: str {
+            constraint exclusive;
+        }
+    }
+
+    type User {
+        required app_id: int32 {
+            constraint exclusive;
+        }
+        required name: str {
+            constraint exclusive;
+        }
+        required email: str {
+            constraint exclusive;
+        }
+        required password: str;
+        client_settings: json;
+
+        badge: Badge;
+        status: Status;
+
+        multi bookmark: Post {
+            note: str;
+        }
+    }
+
+    type Thread {
+        required app_id: int32 {
+            constraint exclusive;
+        }
+        title: str;
+
+        # Let's add a computed link back to posts
+        posts := .<thread[is Post];
+    }
+
+    type Post {
+        required app_id: int32 {
+            constraint exclusive;
+        }
+        required body: str {
+            constraint exclusive;
+        }
+        required creation_time: datetime {
+            default := datetime_current();
+        }
+        edited_time: datetime;
+
+        required user: User;
+        thread: Thread;
+        reply_to: Post;
+    }
+
+
+Copying the Data
+----------------
+
+Now that we have a schema we can use :ref:`ref_cli_edgedb_project_init` and
+:ref:`ref_cli_edgedb_migration` to set up our new EdgeDB database. After the
+schema migration we'll still need to copy the data over the existing data from
+Postgres. JSON is a pretty good intermediate format for this operation. EdgeDB
+can cast data from :eql:type:`json` to all of the built-in scalar types so we
+should be able to use a JSON dump with minimal additional processing when
+importing all the data.
+
+We will dump the ``badges`` and ``statuses`` first:
+
+.. lint-off
+
+.. code-block::
+
+    social=> SELECT ROW_TO_JSON(t) FROM badges AS t;
+                                    row_to_json
+    ---------------------------------------------------------------------------
+     {"name":"admin","description":"Superuser who can do anything"}
+     {"name":"moderator","description":"User who can edit other user's posts"}
+    (2 rows)
+
+    social=> SELECT ROW_TO_JSON(t) FROM statuses AS t;
+            row_to_json
+    ----------------------------
+     {"id":1,"title":"happy"}
+     {"id":2,"title":"sad"}
+     {"id":3,"title":"excited"}
+     {"id":4,"title":"mad"}
+    (4 rows)
+
+.. lint-on
+
+These tables can be dumped directly to a file using a ``COPY ... TO
+<filename>`` command. We can then read the files and use a simple loop to
+import the data into EdgeDB:
+
+.. note::
+
+    When Postgres dumps the JSON data as text files there will be a known
+    gotcha causing all the backslashes used to escape characters inside JSON
+    string values to be doubled. This is because the *text* format causes
+    backslashes themselves to be escaped. This needs to be accounted for when
+    reading the resulting files.
```

Comment:
Here I would make clear that this is Python code as I could see people misreading this as EdgeQL or some other language. Maybe add this:

`Accomplishing this through the Python client library would look like the following:`